src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -81,8 +81,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
     String recipient = request.getParameter("recipient");
+    
+    // Styling text
     text = convertBBCode(text);
-    Message message = new Message(user, text, recipient);
+
+    // Replacing image url to img tag
+    String regex = "(https?://\\S+\\.(png|jpg|gif))";
+    String replacement = "<img src=\"$1\" />";
```

Comment:
Ditto for here, `replacement` is too ambiguous to the reader.

Reply:
replacement is used as an auxiliary string, if any url is found in the text, it will be interpolated in replacement and then appended to the original text so html can display it as an image (or gif 😄 ).